src/site/SiteNavigation.js

```diff
@@ -0,0 +1,207 @@
+import { SetAttribute } from "../utilities/dom.utilities";
+import FocusTrap from "focus-trap";
+
+const attributes = {
+  ariaExpanded: "aria-expanded",
+  tabIndex: "tabindex"
+};
+
+const icons = {
+  open: "fa-bars",
+  close: "fa-times"
+};
+
+const ids = {
+  siteNav: "bc_site-nav",
+  siteNavToggleButton: "bc_site-nav__toggle-button",
+  siteNavLinks: "bc_site-nav__links",
+  page: "bc_page"
+};
+
+const cssClasses = {
+  hidden: "hidden",
+  hiddenInit: "hidden-init",
+  isActive: "is-active",
+  isDisabled: "is-disabled",
+  siteNavToggleButtonText: "bc_site-nav__toggle-button__text"
+};
+
+const invisibleTabIndex = "-1"; /** Allows us to hide an item from tabbing */
+let siteNavFocusTrap;
+
+/**
+ * * Hide/Show the site navigation based on the given shouldShow param
+ * @param {boolean} shouldShow if set to true the site navigation will be shown
+ */
+const toggleSiteNav = shouldShow => {
+  const classListAction = shouldShow ? "add" : "remove";
+  const siteNavToggleButtonElm = document.getElementById(
+    ids.siteNavToggleButton
+  );
+  const buttonIconElm = GetFirstElementOrDefault(siteNavToggleButtonElm, "i");
+  const buttonIconTextElm = GetFirstElementOrDefault(
+    siteNavToggleButtonElm,
+    `.${cssClasses.siteNavToggleButtonText}`
+  );
+  const pageElm = document.getElementById(ids.page);
+
+  // Update toggle button text
+  buttonIconTextElm.textContent = shouldShow ? "Close" : "Menu";
+
+  // Toggle the toggle button icon
+  buttonIconElm.classList.remove(shouldShow ? icons.open : icons.close);
+  buttonIconElm.classList.add(shouldShow ? icons.close : icons.open);
+
+  // Set / Remove aria-expanded attributes
+  SetAttribute(
+    [siteNavToggleButtonElm, pageElm],
+    attributes.ariaExpanded,
+    shouldShow
+  );
+
+  // Set / Remove tab-index inside Site Nav
+  updateSiteNavTabIndex(shouldShow);
+
+  // Hide / Show the site nav
+  document
+    .getElementById(ids.siteNav)
+    .classList[classListAction](cssClasses.isActive);
+
+  // Activate or deactivate proper focus state
+  siteNavFocusTrap[shouldShow ? "activate" : "deactivate"]();
+
+  // Add / Remove esc listener
+  document[shouldShow ? "addEventListener" : "removeEventListener"](
+    "keydown",
+    handleDocumentKeyDown,
+    true
+  );
+
+  // Enable / Disabled page and body
+  pageElm.classList[classListAction](cssClasses.isDisabled);
+  document.body.classList[classListAction](cssClasses.isDisabled);
+};
+
+/**
+ * Handle any click on the document. An if else block will handle any specific
+ * click events we want to capture.
+ * @param {document:click} clickEvent - the observable click event
+ * @listens document:click
+ */
+const handleDocumentClick = clickEvent => {
+  const { target } = clickEvent;
+  const isSiteNavButtonClick =
+    target.id === ids.siteNavToggleButton ||
+    target.closest(`#${ids.siteNavToggleButton}`);
+  const pageElm = target.closest(`#${ids.page}`);
+  const isDisabledPageClick =
```

Comment:
is this using jquery to find the elements? I don't have a problem if it is (i find jquery easier to use to find things) but I thought we were staying away from using it.

Reply:
what part do you think is using jquery?

Reply:
Shoot sorry, yeah I saw # and thought jquery. I see what thats actually doing. 
target.closest(`#${ids.siteNavToggleButton}`); = target.closest("#SomeID") which is just grbabing an element by the ID.